javaslang/src/main/java/javaslang/collection/Traversable.java

```diff
@@ -997,8 +997,10 @@ default T reduceRight(BiFunction<? super T, ? super T, ? extends T> op) {
         }
         if (hasDefiniteSize()) {
             characteristics |= (Spliterator.SIZED | Spliterator.SUBSIZED);
+            return Spliterators.spliterator(iterator(), length(), characteristics);
         }
```

Comment:
Please add `else` here to make it clear that this `if` should stay the last statement:

```java
if (hasDefiniteSize()) {
    ...
    return ...;
} else {
    return ...;
}
```